### PR TITLE
migrate code from googleapis/python-kms

### PR DESCRIPTION
**Pull request by rsamborski**
- Python KMS Apiary P1 samples [(#779)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/779)
- random generation of keyring / cryptokey names [(#786)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/786)
- Auto-update dependencies. [(#790)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/790)
- Remove usage of GoogleCredentials [(#810)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/810)
- Updates Could-KMS sample code to use V1 libraries. [(#856)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/856)
- Remove cloud config fixture [(#887)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/887)
- Re-generate all readmes
- KMS: Clean up base64 logic in the encrypt and decrypt functions. [(#1074)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/1074)
- KMS: Make the samples consistent with the other languages. [(#1075)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/1075)
- Auto-update dependencies. [(#1094)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/1094)
- Update all generated readme auth instructions [(#1121)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/1121)
- Auto-update dependencies. [(#1133)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/1133)
- Added Link to Python Setup Guide [(#1158)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/1158)
- Auto-update dependencies. [(#1186)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/1186)
- Adds snippets for enabling and restoring a key version [(#1196)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/1196)
- Auto-update dependencies. [(#1217)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/1217)
- Added "Open in Cloud Shell" buttons to README files [(#1254)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/1254)
- Auto-update dependencies. [(#1309)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/1309)
- Auto-update dependencies. [(#1320)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/1320)
- Auto-update dependencies. [(#1355)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/1355)
- Auto-update dependencies. [(#1359)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/1359)
- Auto-update dependencies.
- Regenerate the README files and fix the Open in Cloud Shell link for some samples [(#1441)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/1441)
- Update READMEs to fix numbering and add git clone [(#1464)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/1464)
- added kms asymmetric samples [(#1638)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/1638)
- kms text fixes [(#1647)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/1647)
- KMS test fix [(#1690)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/1690)
- KMS changes [(#1723)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/1723)
- KMS import comments [(#1771)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/1771)
- Auto-update dependencies. [(#1846)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/1846)
- Auto-update dependencies. [(#1862)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/1862)
- KMS: Updated for new client library [(#1903)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/1903)
- Auto-update dependencies. [(#1980)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/1980)
- Move import statements into tagged regions [(#2219)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/2219)
- Adds split updates for Firebase ... opencensus [(#2438)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/2438)
- Auto-update dependencies. [(#2005)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/2005)
- chore(deps): update dependency google-cloud-kms to v1.3.0 [(#3160)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/3160)
- Simplify noxfile setup. [(#2806)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/2806)
- [kms] fix flaky test [(#3268)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/3268)
- Update dependency google-cloud-kms to v1.4.0 [(#3410)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/3410)
- chore(deps): update dependency cryptography to v2.9 [(#3266)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/3266)
- fix: use `crypto_key_path` instead of `crypto_key_path_path` [(#3319)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/3319)
- chore(deps): update dependency cryptography to v2.9.1 [(#3463)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/3463)
- [kms] chore: remove gcp-devrel-py-tools [(#3479)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/3479)
- Update dependency cryptography to v2.9.2 [(#3473)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/3473)
- [kms] fix: use unique ids for test [(#3563)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/3563)
- Update and add Cloud KMS samples [(#3690)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/3690)
- chore: some lint fixes [(#3748)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/3748)
- chore: update templates
- chore: update templates
- Add kms/attestations/ directory with sample script to verify attestations [(#3262)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/3262)
- chore(deps): update dependency cryptography to v2.9.2 [(#3666)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/3666)
- Update and add Cloud KMS samples [(#3690)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/3690)
- Add main function to attestation verification script [(#3705)](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/3705)
- test: add noxfile
- chore(deps): update dependency cryptography to v3 (#24)
- feat!: migrate to microgenerator. (#16)
- docs: Generate using new common.py_samples() synthtool functionality (#35)
- chore: update templates (#42)
- chore(deps): update dependency cryptography to v3.1 (#44)
- feat: regenerate client lib to pick up new mtls env (#55)
- chore(deps): update dependency google-cloud-kms to v2 (#29)
- chore(deps): update dependency cryptography to v3.1.1 (#61)
- chore(deps): update dependency google-cloud-kms to v2.2.0 (#57)
- chore(deps): update dependency cryptography to v3.2.1 (#64)
- samples: Add Python code snippets for importing-a-key doc (#67)
- samples: Minor fixes for importing-a-key snippets (#68)
- samples: add request/response integrity verification to encrypt_symmetric.py (#70)
- samples: Add request/response integrity verification to crypto operations (#72)
- chore(deps): update dependency cryptography to v3.3.1 (#78)
- chore(deps): update dependency pem to v21 (#82)
- chore(deps): update dependency cryptography to v3.4.4 (#85)
- chore(deps): update dependency cryptography to v3.4.5 (#88)
- chore(deps): update dependency cryptography to v3.4.6 (#89)
- chore(deps): update dependency cryptography to v3.4.7 (#93)
- chore(deps): update dependency pem to v21.2.0 (#96)
- feat: add script to verify attestations with certificate chains (#99)
- chore(deps): update dependency pytest to v6 (#110)
- chore(deps): update dependency google-cloud-kms to v2.3.0 (#121)
- docs: Include verify_attestation_chains.py help text to attestations README (#134)
- chore(deps): update dependency google-cloud-kms to v2.4.0 (#135)
- chore(deps): update dependency google-cloud-kms to v2.4.1 (#146)
- chore(deps): update dependency google-cloud-kms to v2.4.2 (#153)
- docs: update README for attestation verification scripts (#151)
- chore(deps): update dependency google-cloud-kms to v2.4.3 (#155)
- chore(deps): update dependency google-cloud-kms to v2.5.0 (#159)
- feat(kms): add samples for new hmac and rng apis (#161)
- chore(deps): update dependency cryptography to v3.4.8 (#164)
- chore(deps): update dependency pytest to v6.2.5 (#169)
- chore(deps): update dependency google-cloud-kms to v2.6.0 (#171)
- chore: google-cloud-python now uses 'main' for default branch (#175)
- chore(deps): update dependency google-cloud-kms to v2.6.1 (#180)
- chore(deps): update dependency cryptography to v35 (#182)
- chore(deps): update dependency google-cloud-kms to v2.7.0 (#185)
- chore(deps): update dependency google-cloud-kms to v2.8.0 (#188)
- chore(deps): update dependency google-cloud-kms to v2.9.0 (#194)
- chore(deps): update dependency google-cloud-kms to v2.10.0 (#199)
- chore(deps): update dependency google-cloud-kms to v2.10.1 (#202)
- chore: update README and fix typos in attestation scripts (#203)
- chore(deps): update dependency cryptography to v36 (#210)
- chore(deps): update dependency cryptography to v36.0.1 (#216)
- chore(samples): Add check for tests in directory (#221)
- chore(python): Noxfile recognizes that tests can live in a folder (#225)
- docs(samples): fix typo in verify_asymmetric_ec.py (#227)
- docs(samples): updated var name to avoid shadowing built-in (#238)
- chore(deps): update all dependencies (#234)
- chore: Adding support for pytest-xdist and pytest-parallel (#248)
- chore(deps): update dependency google-cloud-kms to v2.11.1 (#250)
- chore(deps): update dependency pytest to v7.1.0 (#252)
- chore(deps): update all dependencies (#253)
- chore(python): use black==22.3.0 (#256)
- chore(python): add nox session to sort python imports (#267)
- chore(deps): update dependency pytest to v7.1.2 (#270)
- chore(deps): update dependency cryptography to v37 (#271)
- chore(deps): update dependency cryptography to v37.0.1 (#273)
- chore(deps): update dependency cryptography to v37.0.2 (#274)
- fix: require python 3.7+ (#313)
- chore(deps): update all dependencies (#303)
- chore(deps): update all dependencies (#319)
- chore(deps): update dependency google-cloud-kms to v2.12.1 (#326)
- chore(deps): update dependency pytest to v7.1.3 (#338)
- chore(deps): update dependency cryptography to v38 (#339)
- chore(deps): update dependency cryptography to v38.0.1 (#341)
- chore: detect samples tests in nested directories (#345)
- chore(deps): update dependency google-cloud-kms to v2.12.2 (#349)
- chore(deps): update all dependencies (#352)
- chore(deps): update dependency pytest to v7.2.0 (#353)

## Description

Samples migration from https://github.com/googleapis/python-kms

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [x] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [x] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [x] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [x] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [x] Please **merge** this PR for me once it is approved.
- [x] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample

### CODE REVIEW

**Comment by dandhlee** [comment]
Nick, could you help take a look for this migration?

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -65,3 +65,4 @@
 /trace/**/*                            @ymotongpoo @GoogleCloudPlatform/python-samples-reviewers
 /translate/**/*                        @nicain @GoogleCloudPlatform/python-samples-reviewers
 /workflows/**/*                        @GoogleCloudPlatform/python-samples-reviewers
+/kms/**/**                             @GoogleCloudPlatform/dee-infra @GoogleCloudPlatform/python-samples-reviewers
```
Comment (rsamborski) [inline .github/CODEOWNERS:68]: @dandhlee Can we add our team to this repos writers?

Comment (dandhlee) [reply .github/CODEOWNERS:68]: Done.

File: kms/attestations/noxfile.py [hunk 2]
```diff
@@ -0,0 +1,312 @@
+# Copyright 2019 Google LLC
```
Comment (dandhlee) [inline kms/attestations/noxfile.py:1]: Please remove this file, add `noxfile_config.py` file instead.

File: kms/snippets/noxfile.py [hunk 3]
```diff
@@ -0,0 +1,312 @@
+# Copyright 2019 Google LLC
```
Comment (dandhlee) [inline kms/snippets/noxfile.py:1]: Ditto.

File: kms/attestations/noxfile.py (hunk 2 above)
Comment (rsamborski) [reply kms/attestations/noxfile.py]: Done

File: kms/snippets/noxfile.py (hunk 3 above)
Comment (rsamborski) [reply kms/snippets/noxfile.py]: Done

**Review by chalmerlowe — APPROVED** [review]
LGTM

**Review by dandhlee — CHANGES_REQUESTED** [review]
Could you also add `blunderbuss.yml` for `api: cloudkms`?

**Comment by rsamborski** [comment]
> Could you also add `blunderbuss.yml` for `api: cloudkms`?

@dandhlee Done.

File: .github/blunderbuss.yml [hunk 4]
```diff
@@ -126,6 +126,10 @@ assign_issues_by:
   - 'api: monitoring'
   to: 
   - GoogleCloudPlatform/dee-observability
+- labels:
+  - 'api: kms'
```
Comment (dandhlee) [inline .github/blunderbuss.yml:130]: Please use `cloudkms` instead. Python-kms repo used `cloudkms` by the bot and not `kms`.

**Review by dandhlee — CHANGES_REQUESTED** [review]

File: .github/blunderbuss.yml (hunk 4 above)
Comment (rsamborski) [reply .github/blunderbuss.yml:130]: Done. I left kms and added cloudkms as both show up on the labels list in GitHub.

**Review by dandhlee — APPROVED** [review]
Thank you! Overriding CLA and merging.

**Pull request merged**